pkg/controller/networkpolicy/reconciler.go

```diff
@@ -291,6 +292,17 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToPublicNetworks(ctx context.Con
 							carrierGradeNATBlock().String(),
 						}, r.RuntimeNetworks.BlockCIDRs...),
 					},
+				}, {
+					IPBlock: &networkingv1.IPBlock{
+						CIDR: "::/0",
+						// TODO: exclude private blocks so that this actually allows only public networks
```

Comment:
Assign a name to this TODO. Either address it now or explain what needs to be done

Reply:
It's about potentially excluding these networks https://en.wikipedia.org/wiki/Private_network#Private_IPv6_addresses.
Just like you did with IPv4 Networks above. It is more of a question.

Reply:
We tried our best to resolve it here and asked the expert in https://github.com/gardener/gardener/pull/8574/commits/e38b2b67b271ab5ae02022fdfa682c9c56c636b0#r1357015066 :)